pallets/ddc-verification/src/tests.rs

```diff
@@ -1228,3 +1275,283 @@ fn test_get_consensus_nodes_activity_diff_errors() {
+			sent: true,
+			..Default::default()
+		};
+		let pending_request4 = PendingRequest {
+			method: "GET".to_string(),
+			uri: format!("http://{}:{}/activity/era", host4, port),
+			response: Some(era_activity_json4.as_bytes().to_vec()),
+			sent: true,
+			..Default::default()
+		};
+		offchain_state.expect_request(pending_request1);
+		offchain_state.expect_request(pending_request2);
+		offchain_state.expect_request(pending_request3);
+		offchain_state.expect_request(pending_request4);
+
+		drop(offchain_state);
+
+		let node_params1 = StorageNodeParams {
+			ssl: false,
+			host: host1.as_bytes().to_vec(),
+			http_port: port,
+			mode: StorageNodeMode::DAC,
+			p2p_port: 5555,
+			grpc_port: 4444,
+			domain: b"example2.com".to_vec(),
+		};
+
+		let node_params2 = StorageNodeParams {
+			ssl: false,
+			host: host2.as_bytes().to_vec(),
+			http_port: port,
+			mode: StorageNodeMode::DAC,
+			p2p_port: 5555,
+			grpc_port: 4444,
+			domain: b"example3.com".to_vec(),
+		};
+
+		let node_params3 = StorageNodeParams {
+			ssl: false,
+			host: host3.as_bytes().to_vec(),
+			http_port: port,
+			mode: StorageNodeMode::DAC,
+			p2p_port: 5555,
+			grpc_port: 4444,
+			domain: b"example4.com".to_vec(),
+		};
+
+		let node_params4 = StorageNodeParams {
+			ssl: false,
+			host: host4.as_bytes().to_vec(),
+			http_port: port,
+			mode: StorageNodeMode::DAC,
+			p2p_port: 5555,
+			grpc_port: 4444,
+			domain: b"example5.com".to_vec(),
+		};
+
+		let dac_nodes: Vec<(NodePubKey, StorageNodeParams)> = vec![
+			(NodePubKey::StoragePubKey(StorageNodePubKey::new([1; 32])), node_params1),
+			(NodePubKey::StoragePubKey(StorageNodePubKey::new([2; 32])), node_params2),
+			(NodePubKey::StoragePubKey(StorageNodePubKey::new([3; 32])), node_params3),
+			(NodePubKey::StoragePubKey(StorageNodePubKey::new([4; 32])), node_params4),
+		];
+
+		let dac_account = AccountId::from([1; 32]);
+		let cluster_id = ClusterId::from([12; 20]);
+		let era = 16;
+		let merkel_root_hash: H256 = array_bytes::hex_n_into_unchecked(
+			"95803defe6ea9f41e7ec6afa497064f21bfded027d8812efacbdf984e630cbdc",
+		);
+
+		assert_noop!(
+			Pallet::<Test>::get_era_to_validate(cluster_id, dac_nodes.clone()),
+			Error::<Test>::NoValidatorExist
+		);
+
+		ValidatorSet::<Test>::put(vec![dac_account.clone()]);
+
+		assert_ok!(DdcVerification::set_current_validator(RuntimeOrigin::signed(
+			dac_account.clone()
+		),));
+
+		assert_noop!(
+			Pallet::<Test>::get_era_to_validate(cluster_id, dac_nodes.clone()),
+			Error::<Test>::EraToValidateRetrievalError
+		);
+
+		assert_ok!(DdcVerification::create_billing_reports(
+			RuntimeOrigin::signed(dac_account.clone()),
+			cluster_id,
+			era,
+			merkel_root_hash,
+			merkel_root_hash,
+		));
+
+		let result = Pallet::<Test>::get_era_to_validate(cluster_id, dac_nodes.clone());
+		assert!(result.is_ok());
+		assert_eq!(result.unwrap().unwrap(), era_activity2.id); //17
+	});
+}
+
+#[test]
+fn off_chain_worker_works() {
```

Comment:
add test for OCW doesn't work as not enough blocked passed

Reply:
This is covered here https://github.com/Cerebellum-Network/blockchain-node/blob/604d604f52d21ffdb3f7ced833b17e529fbdf310/pallets/ddc-verification/src/tests.rs#L1492-L1497